pyopencl/array.py

```diff
@@ -78,18 +78,7 @@ def _create_vector_types():
 
     counts = [2, 3, 4, 8, 16]
 
-    for base_name, base_type in [
```

Comment:
This points to an important side point--`pyopencl` already has a system for mapping C types to numpy. Why not just tap into that? Search for `get_or_register_dtype`.